### PR TITLE
Document ASO controller settings documentation

### LINKED ISSUE #2202: Document aso-controller-settings valid options
There are various options such as `AZURE_SYNC_PERIOD` which we should document

### PR DESCRIPTION
**Pull request by super-harsh**
Closes #2202

### CODE REVIEW

File: docs/hugo/content/introduction/aso-controller-settings-options.md [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+**Specified As:** `String`
+
+### AZURE_SYNC_PERIOD
+
+AZURE_SYNC_PERIOD is the frequency at which resources are re-reconciled with Azure when
+there have been no triggering changes in the Kubernetes resources.
+This sync exists to detect and correct changes that happened in Azure that Kubernetes is not aware about.
+BE VERY CAREFUL setting this value low - even a modest number of resources can cause
+subscription level throttling if they are re-synced frequently. If nil, sync period defaults to `1h`.
```
Comment (theunrepentantgeek) [inline docs/hugo/content/introduction/aso-controller-settings-options.md:39]:
```suggestion
BE VERY CAREFUL setting this value low - even a modest number of resources can cause
subscription level throttling if they are re-synced frequently. If nil or empty (`""'), sync period defaults to `1h`.
```

File: docs/hugo/content/introduction/aso-controller-settings-options.md [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+**Specified As:** `"1h"`, `"15m"`, or `"60s"`. See https://pkg.go.dev/time#ParseDuration for more details.
+
+### AZURE_OPERATOR_MODE
+
+AZURE_OPERATOR_MODE determines whether the operator should run watchers, webhooks or both(default).
```
Comment (theunrepentantgeek) [inline docs/hugo/content/introduction/aso-controller-settings-options.md:45]:
```suggestion
AZURE_OPERATOR_MODE determines whether the operator should run _watchers_, _webhooks_ or _both_ (default). An empty string, or any unrecognized value, means _both_.
```

File: docs/hugo/content/introduction/aso-controller-settings-options.md [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+
+**Specified As:** `"webhooks"`, `"watchers"` or `"both"`
+
+### AZURE_TARGET_NAMESPACES
+
+AZURE_TARGET_NAMESPACES lists the namespaces the operator will watch for Azure resources (if the mode includes running watchers). 
+If it's empty the operator will watch all namespaces.
+
+**Specified As:** `Comma-separated namespace names`
```
Comment (theunrepentantgeek) [inline docs/hugo/content/introduction/aso-controller-settings-options.md:54]: Does this allow whitespace between values or not?

**Review by theunrepentantgeek — APPROVED** [review]
A few bits of more information to add, but overall looking good.

File: docs/hugo/content/introduction/aso-controller-settings-options.md (hunk 3 above)
Comment (super-harsh) [reply docs/hugo/content/introduction/aso-controller-settings-options.md:54]: it does

**Pull request merged**